### PR TITLE
Development

### PR DESCRIPTION
**Pull request by Castyr**
Added checkstyles, Jacoco (test coverage), added unit tests coverage, and fixed style issues.

### CODE REVIEW

File: src/test/java/org/aist/aide/generaltypeservice/unit/PatternCrudServiceTests.java [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+
+    @Before
+    public void initMocks(){
+        MockitoAnnotations.initMocks(this);
+    }
+
+    @Test
+    public void givenAPatternExists_WhenRequestingThePattern_ThePatternIsReturned() throws NotFoundException {
+        // arrange
+        var type = "Phone";
```
Comment (R4il) [inline src/test/java/org/aist/aide/generaltypeservice/unit/PatternCrudServiceTests.java]: Maybe move these to a class wide final constant, or a class or constants to clean test a bit.(so many Phones)

File: src/test/java/org/aist/aide/generaltypeservice/unit/PatternCrudServiceTests.java [hunk 2]
```diff
@@ -0,0 +1,199 @@
+package org.aist.aide.generaltypeservice.unit;
+
+import org.aist.aide.generaltypeservice.domain.exceptions.NotFoundException;
```
Comment (R4il) [inline src/test/java/org/aist/aide/generaltypeservice/unit/PatternCrudServiceTests.java]: If checkstyle is already active these imports will start having some fun with the order.

**Review by R4il — APPROVED** [review]

**Review by R4il — APPROVED** [review]

File: src/test/java/org/aist/aide/generaltypeservice/integration/TypeResolutionControllerTests.java [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+        typeController
+                .createPattern(
+                        new Pattern("^(\\d{3}-?\\d{2}-?\\d{4}|XXX-XX-XXXX)$",
+                                "SSN"));
+    }
+
+    @After
+    public void tearDown() {
+        var patterns = typeController.getPatterns();
+        for(var pattern: patterns.getBody()) {
```
Comment (R4il) [inline src/test/java/org/aist/aide/generaltypeservice/integration/TypeResolutionControllerTests.java]: lol -checkstyle on for

**Comment by Castyr** [comment]
[Implement General Type Service](https://trello.com/c/Opb1YI22/74-implement-general-type-service)

**Comment by Castyr** [comment]
[Implement General Type Service](https://trello.com/c/Opb1YI22/74-implement-general-type-service)

**Pull request merged**